src/Rings/localization_interface.jl

```diff
@@ -660,5 +660,23 @@ end
 function preimage(f::AbsLocalizedRingHom, I::Ideal)
   base_ring(I) === codomain(f) || error("ideal must be in the codomain of f")
   Q, proj = quo(codomain(f), I)
-  return kernel(compose(f, proj))
+  result = kernel(compose(f, proj))
+  if has_attribute(I, :is_prime) && get_attribute(I, :is_prime) === true
+    set_attribute!(I, :is_prime=> true)
```

Comment:
Don't you want to set the attribute for result instead of 'I'? 'I' already has this attribute (set to true).

Reply:
Yes. Thanks for the catch!